pkg/controller/bypass/bypass_test.go

```diff
@@ -0,0 +1,287 @@
+)
+
+func TestStartByPassKmesh(t *testing.T) {
+	client := fake.NewSimpleClientset()
+
+	err := os.Setenv("NODE_NAME", "test-node")
+	require.NoError(t, err)
+	t.Cleanup(func() {
+		os.Unsetenv("NODE_NAME")
+	})
+
+	addFuncCallCount := 0
+	delFuncCallCount := 0
+
+	patches1 := gomonkey.NewPatches()
+	patches2 := gomonkey.NewPatches()
+	defer patches1.Reset()
+	defer patches2.Reset()
+
+	podWithLabel := &corev1.Pod{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      "test-pod-with-label",
+			Namespace: "default",
+			Labels: map[string]string{
+				"kmesh.net/bypass": "enabled",
+			},
+		},
+		Spec: corev1.PodSpec{
+			NodeName: "test-node",
+		},
+	}
+
+	patches2.ApplyFunc(isKmeshManaged, func(pod *corev1.Pod) bool {
+		return true
+	})
+
+	patches1.ApplyFunc(handleKmeshBypass, func(ns string, oper int) error {
+		if oper == 1 {
+			addFuncCallCount++
+		} else if oper == 0 {
+			delFuncCallCount++
+		}
+		return nil
+	})
+
+	tests := []struct {
+		name       string
+		beforeFunc func()
+		afterFunc  func()
+		wantErr    bool
+	}{
+		{
+			name: "test1: pod with bypass label triggers enable bypass kmesh",
+			beforeFunc: func() {
+				addFuncCallCount = 0
+				delFuncCallCount = 0
+				_, err := client.CoreV1().Pods("default").Create(context.TODO(), podWithLabel, metav1.CreateOptions{})
+				require.NoError(t, err, "failed to create test pod")
+			},
+			afterFunc: func() {
+				assert.Equal(t, 1, addFuncCallCount, "AddFunc should have been called once")
+			},
+			wantErr: false,
+		},
+		{
+			name: "test2: pod without bypass label triggers disable bypass kmesh",
+			beforeFunc: func() {
+				delFuncCallCount = 0
+				err = client.CoreV1().Pods("default").Delete(context.TODO(), "test-pod-with-label", metav1.DeleteOptions{})
+				require.NoError(t, err, "failed to delete test pod")
+			},
+			afterFunc: func() {
+				assert.Equal(t, 1, delFuncCallCount, "DelFunc should have been called once")
+			},
+			wantErr: false,
+		},
+		{
+			name: "test3: pod deletion will not trigger disable bypass kmesh",
+			beforeFunc: func() {
+				addFuncCallCount = 0
+				delFuncCallCount = 0
+				_, err := client.CoreV1().Pods("default").Create(context.TODO(), podWithLabel, metav1.CreateOptions{})
+				require.NoError(t, err, "failed to create test pod")
+				podWithLabel.DeletionTimestamp = &metav1.Time{Time: time.Now()}
+				_, err = client.CoreV1().Pods("default").Update(context.TODO(), podWithLabel, metav1.UpdateOptions{})
+				require.NoError(t, err, "failed to update test pod with deletion timestamp")
+
+				err = client.CoreV1().Pods("default").Delete(context.TODO(), "test-pod-with-label", metav1.DeleteOptions{})
+				require.NoError(t, err, "failed to delete test pod")
+			},
+			afterFunc: func() {
+				assert.Equal(t, 1, addFuncCallCount, "AddFunc should not been called")
+				assert.Equal(t, 0, delFuncCallCount, "DelFunc should have been called once")
+			},
+			wantErr: false,
+		},
+	}
+
+	go StartByPassController(client)
```

Comment:
Why is there a goroutine on this side. Is it possible that the use a goroutine on this side results in a 1s interval between tests?